client/app/budget/budget.controller.js

```diff
@@ -1,174 +1,208 @@
 (function() {
-  'use strict';
-
-  angular
-    .module('travelcal.budget')
-    .controller('BudgetController', BudgetController)
-    .service('CurrencyService', CurrencyService);
-
-
-  function CurrencyService($http) {
-    delete $http.defaults.headers.common['X-Requested-With'];
-
-    var service = {};
-    service.getRates = getRates;
-    service.getCurrencies = getCurrencies;
-    return service;
-
-    function getRates(base) {
-        return $http({
-            method:'Get',
-            url:'http://api.fixer.io/latest?base=' + base
-        }).success(function(data) {
-            return data;
-        }).error(function() {
-            console.log("error");
-        })
-    }
+    'use strict';
+
+    angular
+        .module('travelcal.budget')
+        .controller('BudgetController', BudgetController)
+        .service('CurrencyService', CurrencyService);
+
+
+    function CurrencyService($http) {
+        delete $http.defaults.headers.common['X-Requested-With'];
+        var service = {};
+        service.getRates = getRates;
+        service.getCurrencies = getCurrencies;
+        return service;
+
+        function getRates(base) {
+            return $http({
+                method:'Get',
+                url:'http://api.fixer.io/latest?base=' + base
+            }).success(function(data) {
+                return data;
+            }).error(function() {
+                console.log("error");
+            })
+        }
 
-    function getCurrencies() {
-        return $http({
-            method:'Get',
-            url:'http://api.fixer.io/latest'
-        }).success(function(data) {
-            return data;
-        }).error(function() {
-            console.log("error");
-        })
+        function getCurrencies() {
+            return $http({
+                method:'Get',
+                url:'http://api.fixer.io/latest'
+            }).success(function(data) {
+                return data;
+            }).error(function() {
+                console.log("error");
```

Comment:
`error(function(response) { console.log(response); return response; }` this way a promise will still be returned if error